src/core/Akka.FSharp/Spawning.fs

```diff
@@ -12,129 +12,135 @@ open Microsoft.FSharp.Quotations
+
+    let spawnFromProps (actorFactory : IActorRefFactory) name props =
+        actorFactory.ActorOf(props, name)
+        |> typed
+        :> ActorRef<'Message>
+
+    let createFromExpression actorFactory options name : Linq.Expressions.Expression<Func<'TActor>> -> ActorRef<'Message>=
+        Props.Create
+        >> applySpawnOptions options
+        >> spawnFromProps actorFactory name
+
     /// <summary>
     /// Spawns an actor using specified actor computation expression, using an Expression AST.
     /// The actor code can be deployed remotely.
     /// </summary>
     /// <param name="actorFactory">Either actor system or parent actor</param>
     /// <param name="name">Name of spawned child actor</param>
-    /// <param name="expr">F# expression compiled down to receive function used by actor for response for incoming request</param>
+    /// <param name="actorExpr">F# expression compiled down to receive function used by actor for response for incoming request</param>
     /// <param name="options">List of options used to configure actor creation</param>
-    let spawne (actorFactory : IActorRefFactory) (name : string) 
-        (expr : Expr<Actor<'Message> -> Cont<'Message, 'Returned>>) (options : SpawnOption list) : ActorRef<'Message> = 
-        let e = Linq.Expression.ToExpression(fun () -> new FunActor<'Message, 'Returned>(expr))
-        let props = applySpawnOptions (Props.Create e) options
-        typed (actorFactory.ActorOf(props, name)) :> ActorRef<'Message>
-    
+    let spawne actorFactory options name (actorExpr : Expr<Actor<'Message> -> Cont<'Message, 'Returned>>) : ActorRef<'Message> =
+        (fun () -> FunActor<_,_>(actorExpr))
+        |> Linq.Expression.ToExpression
+        |> createFromExpression actorFactory options name
+
     /// <summary>
     /// Spawns an actor using specified actor computation expression, with custom spawn option settings.
-    /// The actor can only be used locally. 
+    /// The actor can only be used locally.
     /// </summary>
     /// <param name="actorFactory">Either actor system or parent actor</param>
     /// <param name="name">Name of spawned child actor</param>
-    /// <param name="f">Used by actor for handling response for incoming request</param>
+    /// <param name="actor">Used by actor for handling response for incoming request</param>
     /// <param name="options">List of options used to configure actor creation</param>
-    let spawnOpt (actorFactory : IActorRefFactory) (name : string) (f : Actor<'Message> -> Cont<'Message, 'Returned>) 
-        (options : SpawnOption list) : ActorRef<'Message> = 
-        let e = Linq.Expression.ToExpression(fun () -> new FunActor<'Message, 'Returned>(f))
-        let props = applySpawnOptions (Props.Create e) options
-        typed (actorFactory.ActorOf(props, name)) :> ActorRef<'Message>
-    
+    let spawnOpt actorFactory options name (actor : Actor<'Message> -> Cont<'Message, 'Returned>) : ActorRef<'Message> =
```

Comment:
As I've said this is breaking API change, while it's possible advantage is subject to discussion.